specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/privateEndpoint.json

```diff
@@ -1,344 +1,408 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/privateEndpoints": {
+      "get": {
+        "tags": [
+          "PrivateEndpoints"
+        ],
+        "operationId": "PrivateEndpoints_List",
+        "description": "Gets all private endpoints in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of private endpoint resources.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointListResult"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "./network.json#/definitions/Error"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List private endpoints in resource group": { "$ref": "./examples/PrivateEndpointList.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/privateEndpoints": {
+      "get": {
+        "tags": [
+          "PrivateEndpoints"
+        ],
+        "operationId": "PrivateEndpoints_ListBySubscription",
+        "description": "Gets all private endpoints in a subscription.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of private endpoint resources.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointListResult"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "./network.json#/definitions/Error"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all private endpoints": { "$ref": "./examples/PrivateEndpointListAll.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PrivateEndpoint": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointProperties",

```

Comment:
I think you need to add "x-ms-client-flatten": true here

Reply:
Good call. @veronicagg @sarangan12 I believe the linter should have flagged this based on documentation [here](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r2001-avoidnestedproperties)
@bowen0106 please avoid nested properties and apply `x-ms-client-flatten` wherever applicable. Overly nested properties especially in input parameters is bad experience for SDK users